Embedded tutorial

tristantarrant opened the pull request:
Embedded Tutorial
- I have created two site "partials" and a "template" which can be used to add other tutorials
- The tutorial is linked from the "Docs" page
- I have included a script which retags all single commits as steps
- The clustered steps require launching each "node" from a separate terminal. I prefer this approach over launching everything "in one step" since it allows users to experiment with launching on separate boxes
- I am using highlight.js to provide code prettifying
- I am hiding output by default using a collapsed accordion
- The names of the cities in the tutorial are oh-so-random :)

isavin commented on tutorials/embedded/step-0.html.haml:
```diff
@@ -0,0 +1,106 @@
+      %p
+        Let's compile and run it:
+      %pre
+        %code
```
highlightjs detects a keyword here and outputs "package" in bold

%code.nohighlight might be a better option for the command line prompt (even compared to %code.bash which highlights "exec" in the maven cmdline) and the log output (futher down the page in the output section "in" gets highlighted)

isavin commented on tutorials/embedded/step-0.html.haml:
```diff
@@ -0,0 +1,106 @@
+                        Bern, Switzerland - Temperature: -1.6° C, Conditions: broken clouds
+                        London, UK - Temperature: 1.8° C, Conditions: light rain
+                        Newcastle, UK - Temperature: 2.6° C, Conditions: scattered clouds
+                        Bucarest, Romania - Temperature: 9.3° C, Conditions: scattered clouds
```
Small typo, should be "Bucharest".

isavin commented on tutorials/embedded/step-2.html.haml:
```diff
@@ -0,0 +1,112 @@
+                    %code.bash
+                      = preserve do
+                        :escaped
+                          INFO: ISPN000128: Infinispan version: Infinispan 'Hoptimus Prime' 7.1.0.Final
```
Could be removed.

isavin commented on tutorials/embedded/step-3.html.haml:
```diff
@@ -0,0 +1,113 @@
+      = partial( 'tutorialnav.html.haml', { "tutorial_steps" => @tutorial_steps, "tutorial_step" => tutorial_step, "tutorial_github_url" => tutorial_github_url, "header" => true } )
+      %p
+        Typical applications store in the cache data that is the result of an expensive computation or has been retrieved from a slower data-source (e.g. a database or a webservice).
+        If such data is immutable (or mostly immutable), such as names of cities, types of product, etc, it makes sense to store it such as If the data needs to be periodically
```
Small typo "If the data needs" (capital "I").

isavin commented on tutorials/embedded/step-3.html.haml:
```diff
@@ -0,0 +1,113 @@
+                    %code.bash
+                      = preserve do
+                        :escaped
+                          INFO: ISPN000128: Infinispan version: Infinispan 'Hoptimus Prime' 7.1.0.Final
```
Could be removed.

isavin commented on tutorials/embedded/step-6.html.haml:
```diff
@@ -0,0 +1,48 @@
+        In the previous step we successfully started a cluster with two nodes, but we had very little control over the lifecycle of the cluster and both nodes effectively performed
+        the same operations. In this step we will look into cluster events and how they can help in making sure things happen predictably when the cluster topology changes. In order to
+        listen to specific events, we need to create and attach a listener to the CacheManager. An Infinispan listener is simply a class annotated with
+        %i @Listener
```
Small typo, missing "." at the end.

isavin commented on documentation.html.haml:
```diff
@@ -9,7 +9,18 @@ title: Get Started
+          %br
+          %i a tutorial on how to embed Infinispan in your application
+        %td          
+          %a{:href=>"/tutorials/embedded/"} Tutorial
```
It would be nice to have a link pointing to it from the "Getting Started Guide" as well (there are links from there to the User Guide and quickstarts but this is definitely a good starting point as well).

isavin commented on tutorials/embedded/step-7.html.haml:
```diff
@@ -0,0 +1,208 @@
+                          INFO: ISPN000328: Finished local rebalance for cache ___defaultcache on node hoth-1712, topology id = 1
+                          Jan 30, 2015 4:10:41 PM org.infinispan.topology.ClusterTopologyManagerImpl handleRebalanceCompleted
+                          INFO: ISPN000328: Finished local rebalance for cache ___defaultcache on node hoth-9822, topology id = 1
+                          -- Entry for Rome, Italy modified by another node in the cluster
```
Should "Entry for X modified by another node in the cluster" show in the coordinator log as well?

isavin commented on tutorials/embedded/step-7.html.haml:
```diff
@@ -0,0 +1,208 @@
+                    %code.bash
+                      = preserve do
+                        :escaped        
+                          Jan 30, 2015 3:22:08 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport start
```
I think it would be good to drop the ISPN/JGroups log entries completely to make the output more readable.

isavin commented:
Great stuff @tristantarrant !

tristantarrant commented:
@isavin I have incorporated all of your suggestions and I have reworked the docs section by splitting the docs/quickstarts/tutorials and adding links to the config docs.

tristantarrant commented:
Look here http://stg-ispn.rhcloud.com/documentation/

isavin commented:
Integrated and published. Thanks @tristantarrant !
